src/Analyzer.Cli/CommandLineParser.cs

```diff
@@ -154,171 +153,176 @@ private RootCommand SetupCommandLineAPI()
+            this.reportWriter = GetReportWriter(reportFormat, outputFilePath, directoryToAnalyze?.FullName);
+            CreateLoggers(verbose);
 
-            var logger = CreateLogger(verbose, reportFormat, reportWriter);
+            this.templateAnalyzer = TemplateAnalyzer.Create(runPowershell);
 
-            try
+            if (configurationFilePath != null)
             {
-                if (!directoryPath.Exists)
-                {
-                    logger.LogError("Invalid directory: {directoryPath}", directoryPath);
-                    return (int)ExitCode.ErrorInvalidPath;
-                }
-
-                templateAnalyzer.FilterRules(configurationsFilePath);
-
-                // Find files to analyze
-                var filesToAnalyze = new List<FileInfo>();
-                FindJsonFilesInDirectoryRecursive(directoryPath, filesToAnalyze);
+                ConfigurationDefinition config = ReadConfigurationFile(configurationFilePath);
+                if (config == null)
+                    return ExitCode.ErrorGeneric;
 
-                // Log root directory info to be analyzed
-                Console.WriteLine(Environment.NewLine + Environment.NewLine + $"Directory: {directoryPath}");
-
-                int numOfFilesAnalyzed = 0;
-                bool issueReported = false;
-                var filesFailed = new List<FileInfo>();
-                foreach (FileInfo file in filesToAnalyze)
-                {
-                    int res = AnalyzeTemplate(file, null, configurationsFilePath, reportFormat, outputFilePath, runTtk, verbose, false, reportWriter, false, logger);
-                    if (res == (int)ExitCode.Success)
-                    {
-                        numOfFilesAnalyzed++;
-                    }
-                    else if (res == (int)ExitCode.Violation)
-                    {
-                        numOfFilesAnalyzed++;
-                        issueReported = true;
-                    }
-                    else if (res == (int)ExitCode.ErrorGeneric)
-                    {
-                        filesFailed.Add(file);
-                    }
-                }
-
-                Console.WriteLine(Environment.NewLine + $"Analyzed {numOfFilesAnalyzed} file(s).");
-                if (filesFailed.Count > 0)
-                {
-                    logger.LogError($"Unable to analyze {filesFailed.Count} file(s): {string.Join(", ", filesFailed)}");
-                    return (int)(issueReported ? ExitCode.ErrorAndViolation : ExitCode.ErrorGeneric);
-                }
-                return issueReported ? (int)ExitCode.Violation : (int)ExitCode.Success;
-            }
-            catch (Exception exception)
-            {
-                logger.LogError(exception, "An exception occurred while analyzing the directory provided");
-                return (int)ExitCode.ErrorGeneric;
-            }
```

Comment:
I moved a lot of code around, so the diff here isn't great.  This used to be in AnalyzeDirectory.  It's no longer in AnalyzeDirectoryCommandHandler because nothing there is expected to throw an exception (AnalyzeTemplate will catch anything itself and return gracefully).

Reply:
(AnalyzeTemplate will catch anything itself and return gracefully) <-- ok, sounds good.